File: Shared/Data/Models/CharCreator/Body/BodyData.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace TDS.Shared.Data.Models.CharCreator.Body
{
    public class BodyData
    {
        [JsonProperty("99")]
        public byte Slot { get; set; }

        [JsonProperty("0")]
        public virtual List<BodyGeneralData> GeneralDataSynced { get; set; }

        [JsonProperty("1")]
        public virtual List<BodyHeritageData> HeritageDataSynced { get; set; }

        [JsonProperty("2")]
        public virtual List<BodyFeaturesData> FeaturesDataSynced { get; set; }

        [JsonProperty("3")]
        public virtual List<BodyAppearanceData> AppearanceDataSynced { get; set; }

        [JsonProperty("4")]
        public virtual List<BodyHairAndColorsData> HairAndColorsDataSynced { get; set; }
    }
}

File: Client/Window/angular/projects/main/src/app/modules/char-creator/body/models/body-data.ts
import { BodyHeritageData } from './body-heritage-data';
import { BodyFeaturesData } from './body-features-data';
import { BodyAppearanceData } from './body-appearance-data';
import { BodyHairAndColorsData } from './body-hair-and-colors-data';
import { BodyGeneralData } from './body-general-data';

export interface BodyData {
    /** GeneralData */
    [0]: BodyGeneralData[];

    /** HeritageData */
    [1]: BodyHeritageData[];

    /** FeaturesData */
    [2]: BodyFeaturesData[];

    /** AppearanceData */
    [3]: BodyAppearanceData[];

    /** HairAndColorsData */
    [4]: BodyHairAndColorsData[];

    /** Slot */
    [99]: number;
}
